supabasedatapipeline/github_profile_render/ingestor.py

```diff
@@ -65,11 +67,166 @@ def update(self, data):
         for profileData in data:
             print(profileData["github_username"])
             filename = f'{profileData["discord_id"]}githubdisplay.jpg'
-            display = self.getDisplay(profileData)
+            verifiedFileName = f'{profileData["github_username"]}-c4gt-contributions.jpeg'
+            display,verified = self.save_image(profileData)
+            if display != None:
+                try:
+                    self.supabase.client.storage.from_(self.storageBucket).remove(filename)
+                    self.supabase.client.storage.from_("c4gt-github-profile").remove(verifiedFileName)
+
+                except Exception:
+                    pass
+
+                self.supabase.client.storage.from_(self.storageBucket).upload(filename, display, {"content-type": "image/jpeg"})
+                self.supabase.client.storage.from_("c4gt-github-profile").upload(verifiedFileName, verified, {"content-type": "image/jpeg"})
+        
+        return True
+    
+    
+
+    def generateQR(self,border_size=1, ghUsername=''):
+        qr = segno.make_qr(f"https://github-app.c4gt.samagra.io/verify/{ghUsername}")
+        buffer = BytesIO()
+        # Save the QR code to a buffer, adjusting the border size here
+        qr.save(buffer, kind='png', scale=1, border=border_size)
+        qrBytes = buffer.getvalue()
+        return qrBytes
+
+    
+    def save_image(self,data):
+        try:           
+            if data["dpg_points"]+data["classroom_points"]<10:
+                return None,None
+
+            #SET DATA
+            milestone = data["milestone"]
+            prsRaised = data["prs_raised"]
+            prsReviewed = data["prs_reviewed"]
+            prsMerged = data["prs_merged"]
+            dpgPoints = data["dpg_points"]+data["classroom_points"]
+            githubUsername = data["github_username"]
+
+            githubLogoBytes = self.supabase.client.storage.from_('github_profile_assets').download("GHLogoSS.png")
+            githubLogo_np = np.frombuffer(githubLogoBytes, np.uint8)
+            githubLogo = cv2.imdecode(githubLogo_np, cv2.IMREAD_COLOR)
+
+            qrBytes = self.generateQR(1, githubUsername)
+            qr_np = np.frombuffer(qrBytes, np.uint8)
+            qr =  cv2.imdecode(qr_np, cv2.IMREAD_COLOR)
+
+            textDimensions = cv2.getTextSize(githubUsername, cv2.FONT_HERSHEY_DUPLEX, 1.5, 2)
+            usernameWidth = textDimensions[0][0]
+            xCenter = 550
+
+            if githubUsername == "Shruti3004":
+                return None,None
+
+            #GET TEMPLATES
+            if milestone == 1:
```

Comment:
This should be a check using truth values of badge related table columns

Reply:
Im not clear with this comment @KDwevedi  .... pls provide detailed explanations